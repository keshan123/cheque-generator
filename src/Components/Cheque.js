import React from "react";
import PropTypes from "prop-types";

const Cheque = props => (
  <div className="cheque">
    <div className="cheque__payee-and-date-container">
      <div className="cheque__payee">{props.payee}</div>
      <div className="cheque__date-container">
        DATE:
        <div className="cheque__date">
          {props.date.map((number, index) => (
            <div className="cheque__date-individual" key={index}>
              {number}
            </div>
          ))}
        </div>
      </div>
    </div>
    <div className="cheque__text-and-amount-container">
      <div className="cheque__value-text">{props.valueInString}</div>
      {centsDisplay(props.amount) && (
        <div className="cheque__value-cents">{centsDisplay(props.amount)}</div>
      )}
      <div className="cheque__amount">{`$ ${props.amount}`}</div>
    </div>
    <div className="cheque__barcode-and-signature">
      <div className="cheque__bottom-numbers">
        1234:-5678:--..9010::123..-123-522-42
      </div>
      <div className="cheque__signature">Keshan Peiris</div>
    </div>
  </div>
);

const centsDisplay = value => {
  const centsDividable = value.includes(".");
  if (centsDividable) {
    const centsSplit = value.split(".");
    return `${centsSplit[1]} /100`;
  } else {
    return false;
  }
};

Cheque.propTypes = {
  payee: PropTypes.string,
  amount: PropTypes.string,
  date: PropTypes.array,
  valueInString: PropTypes.string
};

export default Cheque;
