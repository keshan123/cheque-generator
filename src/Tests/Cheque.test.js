import React from "react";
import Cheque from "../Components/Cheque";
import renderer from "react-test-renderer";

it("renders correctly without cents", () => {
  const mockPayee = "mockpayee";
  const mockDate = ['1', '2'];
  const mockValueInString = "one hundred dollars";
  const mockAmount = "100";
  const tree = renderer
    .create(
      <Cheque
        payee={mockPayee}
        date={mockDate}
        valueInString={mockValueInString}
        amount={mockAmount}
      ></Cheque>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with cents", () => {
  const mockPayee = "mockpayee";
  const mockDate = ['1', '2'];
  const mockValueInString = "one hundred dollars";
  const mockAmount = "100.20";
  const tree = renderer
    .create(
      <Cheque
        payee={mockPayee}
        date={mockDate}
        valueInString={mockValueInString}
        amount={mockAmount}
      ></Cheque>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
