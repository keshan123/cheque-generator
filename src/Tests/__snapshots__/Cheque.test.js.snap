// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with cents 1`] = `
<div
  className="cheque"
>
  <div
    className="cheque__payee-and-date-container"
  >
    <div
      className="cheque__payee"
    >
      mockpayee
    </div>
    <div
      className="cheque__date-container"
    >
      DATE:
      <div
        className="cheque__date"
      >
        <div
          className="cheque__date-individual"
        >
          1
        </div>
        <div
          className="cheque__date-individual"
        >
          2
        </div>
      </div>
    </div>
  </div>
  <div
    className="cheque__text-and-amount-container"
  >
    <div
      className="cheque__value-text"
    >
      one hundred dollars
    </div>
    <div
      className="cheque__value-cents"
    >
      20 /100
    </div>
    <div
      className="cheque__amount"
    >
      $ 100.20
    </div>
  </div>
  <div
    className="cheque__barcode-and-signature"
  >
    <div
      className="cheque__bottom-numbers"
    >
      1234:-5678:--..9010::123..-123-522-42
    </div>
    <div
      className="cheque__signature"
    >
      Keshan Peiris
    </div>
  </div>
</div>
`;

exports[`renders correctly without cents 1`] = `
<div
  className="cheque"
>
  <div
    className="cheque__payee-and-date-container"
  >
    <div
      className="cheque__payee"
    >
      mockpayee
    </div>
    <div
      className="cheque__date-container"
    >
      DATE:
      <div
        className="cheque__date"
      >
        <div
          className="cheque__date-individual"
        >
          1
        </div>
        <div
          className="cheque__date-individual"
        >
          2
        </div>
      </div>
    </div>
  </div>
  <div
    className="cheque__text-and-amount-container"
  >
    <div
      className="cheque__value-text"
    >
      one hundred dollars
    </div>
    <div
      className="cheque__amount"
    >
      $ 100
    </div>
  </div>
  <div
    className="cheque__barcode-and-signature"
  >
    <div
      className="cheque__bottom-numbers"
    >
      1234:-5678:--..9010::123..-123-522-42
    </div>
    <div
      className="cheque__signature"
    >
      Keshan Peiris
    </div>
  </div>
</div>
`;
